docs/source/trainer.rst

```diff
@@ -157,6 +157,51 @@ Once you're done training, feel free to run the test set!
 
     trainer.test(test_dataloaders=test_dataloader)
 
+Saving test predictions is simple ! You can provide directly a tensor or list of dictionaries.
+
+.. code-block:: python
+
+    import pandas as pd
+
+    class MyModel(LightningModule):
+
+        def test_step(self, batch, batch_idx, dataloader_idx=None):
+            x, y = batch
+            output = self.layer(x)
+            loss = self.loss(y, output)
+
+            # Add directly a tensor
+            self.add_predictions(output)
+
+            return {"y": loss}
+
+    results = trainer.test(test_dataloaders=test_dataloaders)
+    for dataloader_idx, dataloader_result in enumerate(results_:
+
+        df = pd.json_normalize(dataloader_result["predictions"], sep='_')
+
+.. code-block:: python
+
+    import pandas as pd
+
+    class MyModel(LightningModule):
+
+        def test_step(self, batch, batch_idx, dataloader_idx=None):
+            x, y = batch
+            output = self.layer(x)
+            loss = self.loss(y, output)
+
+            # Support list of dictionaries.
+            self.add_predictions([{"pred": o, "target": t} for o, t in zip(output, y)])
+
+            return {"y": loss}
+
+    results = trainer.test(test_dataloaders=test_dataloaders)
+    for dataloader_idx, dataloader_result in enumerate(results_:
+
+        df = pd.json_normalize(dataloader_result["predictions"], sep='_')
```

Comment:
strictly speaking, this example will overwrite `df` for each new dataloader. Could we concatenate the `df` from each dataloader or something (i know this assumes that predictions have the same size for all dataloaders)

Reply:
Done !